microcosm_postgres/encryption/registry.py

```diff
@@ -11,23 +11,32 @@
 
 from microcosm_postgres.encryption.encryptor import MultiTenantEncryptor, SingleTenantEncryptor
 from microcosm_postgres.encryption.providers import (
-    configure_key_provider,
+    configure_decrypting_key_provider,
+    configure_encrypting_key_provider,
     configure_materials_manager,
 )
 
 
 def parse_config(context_keys: Sequence[str],
-                 key_ids: Sequence[str]) -> Mapping[str, Sequence[str]]:
+                 key_ids: Sequence[str],
+                 account_ids: Sequence[str],
+                 partitions: Sequence[str]) -> Mapping[str, Mapping[str, Sequence[str]]]:
     return {
         # NB: split key id on non-comma to avoid confusion with config parsing
-        context_key: key_id.split(";") if isinstance(key_id, str) else key_id
-        for context_key, key_id in zip(context_keys, key_ids)
+        context_key: {
+            "key_ids": key_id.split(";") if isinstance(key_id, str) else key_id,
+            "account_ids": account_id.split(";") if isinstance(account_id, str) else account_id,
+            "partition": partition,
+        }
+        for context_key, key_id, account_id, partition in zip(context_keys, key_ids, account_ids, partitions)
```

Comment:
The use of `zip` here does make this backwards incompatible, unfortunately. I could loosen this up a bit to allow empty partition/account_ids to be passed in.